### PR TITLE
VS Code Marketplace Last Updated Badge added along with tests, run [VisualStudioMarketplace]

### PR DESCRIPTION
**Pull request by YashTotale**
## New badge for the VS Code Marketplace: Last Updated
Preview

_Tests Included_

### CODE REVIEW

**Comment by calebcartwright** [comment]
This looks to also contain the Release Date badge commits from #5387. These two should be sufficiently small to address in a single PR, so I am going to close #5387 and track the addition of both within this PR. If you'd still rather have them as separate PRs, then please be sure to make them truly distinct/remove duplicative changes and commits

**Comment by YashTotale** [comment]
> This looks to also contain the Release Date badge commits from #5387. These two should be sufficiently small to address in a single PR, so I am going to close #5387 and track the addition of both within this PR. If you'd still rather have them as separate PRs, then please be sure to make them truly distinct/remove duplicative changes and commits

Yeah, I think having both in a single PR would be better.

File: services/visual-studio-marketplace/visual-studio-marketplace-base.js [hunk 1]
```diff
@@ -27,6 +27,8 @@ const extensionQuerySchema = Joi.object({
                 )
                 .min(1)
                 .required(),
+              releaseDate: Joi.string().required(),
+              lastUpdated: Joi.string().required(),
```
Comment (calebcartwright) [inline services/visual-studio-marketplace/visual-studio-marketplace-base.js:31]: Are both of these field guaranteed to always be there for all packages, and never be null or undefined? `releaseDate` feels like a safe bet, but want to make sure that `lastUpdated` is there immediately after the initial publish and not just null until the first _update_

File: services/visual-studio-marketplace/visual-studio-marketplace-last-updated.tester.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            ],
+          },
+        ],
+      })
+  )
+  .expectBadge({
+    label: 'last updated',
+    message: 'april 2013',
+    color: 'red',
+  })
```
Comment (calebcartwright) [inline services/visual-studio-marketplace/visual-studio-marketplace-last-updated.tester.js:50]: This looks like a mocked version of the live test above, so let's remove it. Our service tests are largely a suite of integration tests with the myriad upstream services and platforms Shields integrates with, so we try to be mindful about having duplicative tests within here. 

I know some of the other VS Marketplace service tests have similar ones, but we don't need to mimic that here.

File: services/visual-studio-marketplace/visual-studio-marketplace-last-updated.tester.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    label: 'last updated',
+    message: 'extension not found',
+  })
+
+t.create('invalid')
+  .get('/visual-studio-marketplace/last-updated/yasht.terminal-all-in-one.json')
+  .intercept(nock =>
+    nock('https://marketplace.visualstudio.com/_apis/public/gallery/')
+      .post(`/extensionquery/`)
+      .reply(401)
```
Comment (calebcartwright) [inline services/visual-studio-marketplace/visual-studio-marketplace-last-updated.tester.js:73]: What circumstances result in us receiving a 401 response status code?

File: services/visual-studio-marketplace/visual-studio-marketplace-release-date.tester.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            ],
+          },
+        ],
+      })
+  )
+  .expectBadge({
+    label: 'release date',
+    message: 'april 2013',
+    color: 'red',
+  })
```
Comment (calebcartwright) [inline services/visual-studio-marketplace/visual-studio-marketplace-release-date.tester.js:50]: Same comment as the similar test in the last updated service tester file, let's remove this

```suggestion
```

File: services/visual-studio-marketplace/visual-studio-marketplace-release-date.tester.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    label: 'release date',
+    message: 'extension not found',
+  })
+
+t.create('invalid')
+  .get('/visual-studio-marketplace/release-date/yasht.terminal-all-in-one.json')
+  .intercept(nock =>
+    nock('https://marketplace.visualstudio.com/_apis/public/gallery/')
+      .post(`/extensionquery/`)
+      .reply(401)
```
Comment (calebcartwright) [inline services/visual-studio-marketplace/visual-studio-marketplace-release-date.tester.js:73]: Same question here on circumstances that cause a 401

File: services/visual-studio-marketplace/visual-studio-marketplace-base.js (hunk 1 above)
Comment (YashTotale) [reply services/visual-studio-marketplace/visual-studio-marketplace-base.js:31]: Yes, `lastUpdated` is there right after the initial publish. You can try it out using this extension I found with only 1 release: `tscpp.node-dependencies`.

File: services/visual-studio-marketplace/visual-studio-marketplace-last-updated.tester.js (hunk 2 above)
Comment (YashTotale) [reply services/visual-studio-marketplace/visual-studio-marketplace-last-updated.tester.js:50]: Got it! It's removed.

File: services/visual-studio-marketplace/visual-studio-marketplace-release-date.tester.js (hunk 4 above)
Comment (YashTotale) [reply services/visual-studio-marketplace/visual-studio-marketplace-release-date.tester.js:50]: Removed

File: services/visual-studio-marketplace/visual-studio-marketplace-last-updated.tester.js (hunk 3 above)
Comment (YashTotale) [reply services/visual-studio-marketplace/visual-studio-marketplace-last-updated.tester.js:73]: Removed 401 tests

File: services/visual-studio-marketplace/visual-studio-marketplace-release-date.tester.js (hunk 5 above)
Comment (YashTotale) [reply services/visual-studio-marketplace/visual-studio-marketplace-release-date.tester.js:73]: Removed 401 tests.

**Pull request merged**